featuretools/entityset/entityset.py

```diff
@@ -1265,7 +1302,7 @@ def _check_uniform_time_index(self, dataframe, column_id=None):
     def _get_time_type(self, dataframe, column_id=None):
         column_id = column_id or dataframe.ww.time_index
 
-        # --> try and find a way to test this case!!!!
+        # --> TODO try and find a way to test this case!!!!
         if dataframe.ww.schema is None:
             column_schema = dataframe[column_id].head().ww.init()
```

Comment:
Okay, so looking at how we use `_get_time_type` internal to featuretools, we should be fine. 

My one concern is external users of `EntitSet.set_secondary_time_index` which takes in a dataframe and passes that to `_check_secondary_time_index` where we eventually do the time type check. Internally, when we call it, we only pass Woodwork DataFrames, but would a user maybe call it on a random dataframe? (It wouldn't end up working anyway because we update the Woodwork metadata, so either we need to specifically disallow it or initialize Woodwork on the DataFrame if it's not already initialized).

No matter what, though, it seems like we shouldn't need to account for this case in `_get_time_type`.

Reply:
Ah - I think we may need to update `set_secondary_time_index`. I don't think that should take in an arbitrary dataframe - it should take in a dataframe id, and pull the dataframe from the EntitySet.

Reply:
That change makes sense to me!

Reply:
Although, I'm not sure how that flow would work in `add_dataframe`...

Reply:
hmm... I can't currently see any problems with just moving up the `self.dataframe_dict[dataframe_id] = dataframe` line so that we deal with secondary time index after adding it into the EntitySet, but maybe I'm missing something here

Reply:
It might not be a big deal, but I think if there is an error during the setting the secondary time index, the dataframe will still be part of the EntitySet at that point, but not quite configured the same as requested with respect to the secondary time index.

Reply:
Yeah, that's probably a fair reason to not rearrange the order here. I don't want to overcomplicate things, so maybe we just confirm that the dataframe has Woodwork initialized in `set_secondary_time_index` and if it's not, raise an error.

Alternatively, if we really don't want users to be passing in dataframes, this can be a helper method `_set_secondary_time_index` that takes in a dataframe (should probably still confirm that Woodwork is initialized), and then if we want a public method, we can allow a `set_secondary_time_index` that just takes in a dataframe id and grabs the dataframe, passing it to the private method.

Reply:
I like the idea of having a public method that expects a string id and then a private method that takes a dataframe that we can use in `EntitySet.add_dataframe`.

@rwedge Any thoughts on this?

Reply:
added this in along with some new tests. @rwedge happy to change anything if you have any additional thoughts!